vsintegration/src/FSharp.ProjectSystem.FSharp/Project.fs

```diff
@@ -355,6 +355,14 @@ namespace rec Microsoft.VisualStudio.FSharp.ProjectSystem
                 pIdIco <- 400u
                 VSConstants.S_OK
 
+            override this.Dispose(disposing) =
```

Comment:
It doesn't look thread safe.  But there may be other mitigations in place.  

However, as it stands two threads could both invoke mgr..FRevokeComponent(componentID) simultaneously.

Now that might not be a problem ... 
  for example if mgr..FRevokeComponent(componentID) is resilient to double revoke.

or
 Something higher up the stack is ensuring that two threads can't dispose simultaneous ... for example something enforces that only a single thread at a time is doing the dispose.

to make it threadsafe either grabbing a lock and double checking or using Interlocked.CompareExchange to set componentID back to zero would probably work fine.